### ISSUE TITLE
Usage of stochastic duration predictor

### ISSUE THREAD

**Issue by mrakotos** [shivammehta25/Matcha-TTS#125]
Can we replace the duration predictor with the stochastic duration predictor of vits.

**Comment by shivammehta25** [comment]
Yes! But even better, use Matcha's own stochastic duration predictor :) on this branch: https://github.com/shivammehta25/Matcha-TTS/tree/stoc_dur

**Comment by Marioando** [comment]
how about the performance ?

**Comment by mrakotos** [comment]
Hi , is this the same flow matching duration predictor as in [this paper](https://www.isca-archive.org/interspeech_2024/mehta24b_interspeech.pdf).

**Comment by shivammehta25** [comment]
Yes! it is :)

**Comment by lumpidu** [comment]
Sorry for the late capture: have you compared the FM based duration predictor with the SDP from VITS ? I am currently going the other way around and try your Flow matching implementation as a replacement for VITS SDP. Unfortunately, you haven't made any comparison in your paper between both - only DDP and FDP.

I am in general relatively happy with the SDP in VITS, but to improve its performance for longer texts and emotional speech, I want to try different experiments. There are of course more options than to replace the architecture itself, e.g. using a better text encoder, which again can give better features to SDP/FDP, external alignments instead of MAS, etc.

In general, I wonder if you have seen any form of mode collapse with your implementation? I can see that you use MSE loss, but no penalty for if the model learns to predict only mean durations.
